### PR TITLE
Update create remote client

### LINKED ISSUE #1117: Update Add remote connection page to use Ant Design
## What needs changed?

Full refactor with Ant Design

## How does it work now?

How does this component work now, and what's wrong with it?

## How should it work?

How do you think this component should work?

## Additional information

Add any other context, files, or screenshots about this request here.

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

Updated the create remote client to now be in an Ant Design modal window.  Functions exactly the same as before.

## Related issue

Fixes #1117

## Checklist

Things for the developer to confirm they've done before the PR should be accepted:

*   [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
*   [x] Tests added (or description of how to test) for any new features.
*   [x] User documentation updated for UI or technical changes.

### CODE REVIEW

**Review by ksierks — CHANGES_REQUESTED** [review]
I took a peek all the files that aren't tests so far. Sorry for all the questions.

File: src/main/webapp/resources/js/pages/remote-apis/RemoteApiPage.jsx [hunk 1]
```diff
@@ -13,21 +14,20 @@ import { PagedTableProvider } from "../../components/ant.design/PagedTable";
...
-      }
-    >
-      <PagedTableProvider url={setBaseUrl("ajax/remote_api/list")}>
+    <PagedTableProvider url={setBaseUrl("ajax/remote_api/list")}>
+      <PageWrapper
+        title={i18n("RemoteApi.title")}
+        headerExtras={
+          window.PAGE.admin ? (
+            <CreateRemoteApiModal>
+              <AddNewButton text={i18n("remoteapi.add")} />
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/remote-apis/RemoteApiPage.jsx:23]: As a child?  It allows the modal to use anything as the toggle for opening.  Just makes more reusable code.  I have been trying to use this pattern more and more.

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,6 +43,7 @@ Changes
 * [Database]: Updated query to improve performance for metadata in REST API and line list.
 * [UI]: Fixed bug when exporting samples from project > samples page while using filter by file, all sample were exported.
 
+
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply CHANGELOG.md]: Cleaned up when I merged development

File: src/main/resources/i18n/messages.properties [hunk 3]
```diff
@@ -1311,10 +1311,17 @@ RemoteApi.disconnected=Connect
...
 remoteapi.add=Add Remote Connection
 remoteapi.name=Name
+remoteapi.name.required=A name is required for a Remote Connection
 remoteapi.clientid=Client ID
+remoteapi.clientid.required=A client id is required
+remoteapi.clientid.no-spaces=A client ID cannot contain any spaces
 remoteapi.serviceurl=Service URL
+remoteapi.serviceurl.required=A service URL is required
+remoteapi.serviceurl.url=Must be a valid url
 remoteapi.details.title=Remote Connection Details
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/resources/i18n/messages.properties]: I think I got them all, thanks

File: src/main/webapp/resources/js/pages/remote-apis/RemoteApiPage.jsx [hunk 4]
```diff
@@ -13,21 +15,20 @@ import { PagedTableProvider } from "../../components/ant.design/PagedTable";
...
-      }
-    >
-      <PagedTableProvider url={setBaseUrl("ajax/remote_api/list")}>
+    <PagedTableProvider url={setBaseUrl("ajax/remote_api/list")}>
+      <PageWrapper
+        title={i18n("RemoteApi.title")}
+        headerExtras={
+          isAdmin() ? (
+            <CreateRemoteApiModal>
+              <AddNewButton text={i18n("remoteapi.add")} />
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/remote-apis/RemoteApiPage.jsx]: Updated in d141d33

File: src/main/webapp/resources/js/pages/admin/components/remote-connections/RemoteApiStatus.jsx [hunk 5]
```diff
@@ -36,7 +36,7 @@ export function RemoteApiStatus({ api, onConnect = () => {} }) {
       setLoading(false);
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/admin/components/remote-connections/RemoteApiStatus.jsx]: Updated in d141d33

**Review by ksierks — APPROVED** [review]
Looks good to me.

**Pull request merged**